lab_6_pipeline/pipeline.py

```diff
@@ -18,11 +27,26 @@ def __init__(self, path_to_raw_txt_data: Path):
         """
         Initializes CorpusManager
         """
+        self.path_to_raw_txt_data = Path(path_to_raw_txt_data)
+        self._storage = {}
+        self._validate_dataset()
+        self._scan_dataset()
 
     def _validate_dataset(self) -> None:
         """
         Validates folder with assets
         """
+        if not self.path_to_raw_txt_data.exists():
+            raise FileNotFoundError
+        if not self.path_to_raw_txt_data.is_dir():
+            raise NotADirectoryError
+        if not any(self.path_to_raw_txt_data.iterdir()):
+            raise EmptyDirectoryError
+        meta = list(self.path_to_raw_txt_data.glob("*_meta.txt"))
+        raw = list(self.path_to_raw_txt_data.glob("*_raw.json"))
+        if len(meta) != len(raw):
+            raise InconsistentDatasetError
```

Comment:
consider also raising this error when the numeration of the files in the dataset is not right